### ISSUE TITLE
[Enhancement proposal]: Permission Table Documentation

### ISSUE THREAD

**Issue by TheFinestArtist** [Baseflow/flutter-permission-handler#1204]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-permission-handler/issues).
- [X] I have carefully [read the documentation](https://github.com/baseflow/flutter-permission-handler/blob/main/permission_handler/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [X] iOS
- [X] Windows

### Proposal

![Untitled](https://github.com/Baseflow/flutter-permission-handler/assets/3123741/62cef900-c8ba-4a52-aa6e-185f7b019a10)

I was collecting information regarding permissions that I need to support the app I am working on. I think it would be amazing for this library to support such information as a table.

Example Table
| Permission | OS | Supporting API | Related Doc | Permission Name | Permission Status | 
| :---: | :---: | :---: | :---: | :---: | :---: |
| `Notification` | Android | API Level 33 or higher | [Doc](https://developer.android.com/develop/ui/views/notifications/notification-permission) | `notification` | `granted, denied, permanentlyDenied` |
| `Do Not Disturb` | Android | API Level 23 or higher | [Doc](https://developer.android.com/reference/android/app/NotificationManager#setInterruptionFilter(int)) | `accessNotificationPolicy` | `granted, denied` |

### Pitch

iOS, Android, Windows, etc permissions is one of the most confusing features to support for not only engineers but product managers or even designers. It would be a great contribution to the community if there is a single source that accumulates all the information in one place. The whole front-end development community will be beneficial out of this.

**Comment by JeroenWeener** [comment]
Hi @TheFinestArtist, thank you for bringing this up.

We agree that a table that provides an overview of the permissions and what they reflect is desirable. We are currently working on a major refactor of the plugin, which will, amongst other things, change how platform specific permissions are accessed. We decided that it is probably best to wait for the refactor to land before publishing such a table, as the refactor will change a substantial amount of the info that will be in the table. I will mark this issue as `refactor`, so we get reminded about it once the refactor lands.

**Comment by TheFinestArtist** [comment]
Sounds like a plan, Jeroen! I will look forward to the new big update and revisiting this documentation after then.
